pkg/network/ebpf/c/protocols/http2/decoding.h

```diff
@@ -326,7 +327,16 @@ static __always_inline void parse_frame(struct __sk_buff *skb, skb_info_t *skb_i
         skb_info->data_off += current_frame->length;
     }
 
-    if ((current_frame->flags & HTTP2_END_OF_STREAM) == HTTP2_END_OF_STREAM) {
+    // When we accept an RST with NO ERROR, it means that the current stream was terminated with no error.
+    // According to https://datatracker.ietf.org/doc/html/rfc7540#section-6.4, the frame includes a 32-bit
+    // error code integer.
+    if (current_frame->type == kRSTStreamFrame) {
+        __u32 rst_error_code = 0;
+        bpf_skb_load_bytes(skb, skb_info->data_off, &rst_error_code, sizeof(rst_error_code));
+        is_rst_rst_with_no_error_code = rst_error_code == HTTP2_RST_NO_ERROR;
+    }
```

Comment:
rethinking about that, 
if we have RST on the stream, we shouldn't care for the reason (error/no-error)

Reply:
Changed